packages/convertkit/src/forms/subscribe-to-convertkit/index.tsx

```diff
@@ -66,76 +35,43 @@ export const redirectUrlBuilder = (
  * @constructor
  */
 export const SubscribeToConvertkitForm: React.FC<SubscribeFormProps> = ({
-  formId = CONVERTKIT_SIGNUP_FORM,
+  formId,
   submitButtonElem,
   errorMessage = <p>Something went wrong.</p>,
   successMessage = <p>Thanks!</p>,
   actionLabel = 'Subscribe',
   onError = () => {},
   onSuccess = () => {},
-  subscribeApiURL = CONVERTKIT_SUBSCRIBE_API_URL,
+  subscribeApiURL,
   id,
   ...rest
 }) => {
-  const [isSubmitting, setSubmitting] = React.useState<boolean>(false)
```

Comment:
formik actually tracks this state BUT your on submit handler has to return async value (promise)

https://formik.org/docs/api/formik#onsubmit-values-values-formikbag-formikbag--void--promiseany

below `handleOnSubmit` is async, but doesn't return (so it will be `undefined` I think?)

Reply:
good catch 👍